src/browser/core_state.ts

```diff
@@ -516,6 +521,15 @@ export function getAppAncestor(descendantAppUuid: string): Shapes.App {
     }
 }
 
+function getExternalAncestor(descendantAppUuid: string): any {
```

Comment:
Can we specify shape here instead of `any`?

Reply:
We need to create shapes for external connection